### PR TITLE
Enhancement: Work continued.

### PR DESCRIPTION
**Pull request by nmn**
Taking the work put in by @Adezandee I have added some stuff to make it better:

1. Added more types in all the files being used.
2. Added eslint etc. to package.json as devDependencies. (I see no downside in helping contributors install these dependencies)
3. Reverted to using `check-contents` which, once again, enables on-the-fly linting.

Note:
The `enableAll` configuration now, is essentially just a test of whether to look for the `@flow` comment or not. You can ask flow to type-check any file by passing it using `check-contents` even if it doesn't have the `@flow` comment. So it essentially works the same way.

That said, the linter will only report errors for the currently active file and not the whole project, as that is a limitation of `check-contents`. This is the correct behaviour as the scope of the linter is set to `file` and not `project`.

I have tested this locally and everything works as expected. I'm not a fan of the notifications on start-up, but I'm OK with it.

@Adezandee My understanding is that your PR is part of this PR. Am I correct? In that case, can we close the other PR?

### CODE REVIEW

**Comment by steelbrain** [comment]
I have a question, he opened a PR on this very repo, couldn't you just push to it and make the changes you want to do there, instead of opening a new PR? :smiley:

**Comment by nmn** [comment]
@steelbrain I tried to do that, but for some reason, github won't let me.

**Comment by nmn** [comment]
Hold on, let me try again.

**Comment by steelbrain** [comment]
Just a side note that here's what I would do if I wanted to do that,
```
cd ~/.atom/packages
rm -rf linter-flow
git clone --depth=1 git@github.com:AtomLinter/linter-flow.git
git checkout -b adezandee/flow-server
git pull -u origin adezandee/flow-server
```

That said, both of you are Geniuses, and you would be very productive working together in one PR instead of two

**Comment by nmn** [comment]
Closed, as I managed to push the changes to @Adezandee 's PR. Sorry for the confusion.
